pkg/controller/infrastructure/actuator_delete.go

```diff
@@ -51,10 +51,14 @@ func Delete(
 		return fmt.Errorf("could not create the Terraformer: %+v", err)
 	}
 
+	// terraform pod from previous reconciliation might still be running, ensure they are gone before doing any operations
+	if err := tf.EnsureCleanedUp(ctx); err != nil {
```

Comment:
I am wondering, if we should rather use `WaitForCleanEnvironment` here to not interrupt any running `tf-destroy` pods if the deletion is triggered again (backoff on provider-aws side / timeout and retry on gardenlet side / ...)
The benefit of `EnsureCleanup` would be, that running `tf-apply` pods would be stopped directly instead of waiting for high timeouts.

Reply:
Hmm, good question. My personal gud feeling would be that `WaitForCleanEnvironment` is a better starting point. Yes, we might potentially run into the timeout, but I cannot overlook the side effects that the termination of a running `destroy` pod could have.
Generally, PR lgtm, thanks.

Reply:
Ok, thanks for your opinion. Then let's go for `WaitForCleanEnvironment` 👍